### ISSUE TITLE
JIT Compile: std::bad_alloc at memory location 

### ISSUE THREAD

**Issue by deepakkumar1984** [pytorch/pytorch#24228]
## 🐛 Bug

Trying to run this code as per the documentation

```
#include <torch/torch.h>

int main()
{
	auto module = torch::jit::compile(R"JIT(
					def relu_script(a, b):
					return torch.relu(a + b)
					def test_while(a, i):
					while i < 10:
						a += a
						i += 1
					return a
				)JIT");
	
	auto func = module->find_function("relu_script");
	int exit;
	std::cin >> exit;
	return 0;
}
```

But failing with error below:

![image](https://user-images.githubusercontent.com/6328361/62915384-66aeed80-bdd3-11e9-999d-bf612f66bba1.png)

 - PyTorch Version (e.g., 1.0): 1.2
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): c++
 - Build command you used (if compiling from source): cmake
 - Python version: NA
 - CUDA/cuDNN version: 10, 7.5
 - GPU models and configuration: NVIDIA 1080Ti
 - Any other relevant information:

cc @suo

**Comment by driazati** [comment]
Can you post a full stack trace? Aside from some indentation errors (below), this example doesn't produce the same error for me on master with Linux.

`return ...` and `while ...`
```cpp
#include <torch/torch.h>

int main()
{
	auto module = torch::jit::compile(R"JIT(
          def relu_script(a, b):
            return torch.relu(a + b)
          def test_while(a, i):
            while i < 10:
              a += a
              i += 1
            return a
        )JIT");

	auto func = module->find_function("relu_script");
	return 0;
}```

**Comment by deepakkumar1984** [comment]
I downloaded the pytorch debug package and it works fine. Unable to work with the release package v1.2. Used the nightly build, same issue. I am not able to give you detailed exception stack.
I noticed its getting failed at kernalbase.dlll

![image](https://user-images.githubusercontent.com/6328361/62988064-13e03f00-be82-11e9-86ab-8a32d01ccfee.png)

![image](https://user-images.githubusercontent.com/6328361/62988077-1e9ad400-be82-11e9-8464-f7007c122659.png)


If the above screesnhot helps
